rfcs/0051-stop-transaction-processing.md

```diff
@@ -0,0 +1,203 @@
+# Stop processing transactions / stop the network after a certain slot
+
+This PR describes the feature to stop processing transactions and to stop the
+network after a certain slot, to be used in the Berkeley hard fork.
+
+## Summary
+
+Transactions come from a client or the gossip network and are processed by BPs
+and SNARK workers to be included in blocks. These blocks are propagated through
+the network and validated by other participants.
+
+In this RFC, the procedure to stop processing any new transactions and to
+stop the network after a certain slot is described. This is, we define two
+slots: the first one is the slot after which any blocks produced will include no
+transaction at all and no fee payments, and the second one is the slot after
+which no blocks are produced and blocks received are rejected.
+
+## Motivation
+
+In a hard fork scenario, we want to halt the preceding network and produce a new
+genesis ledger for the succeeding network. This new genesis ledger should be
+produced from a stabilised staged ledger from the preceding network. This is, we
+define a point in time (slot) where the network continues to operate but with no
+"activity". In detail, after this slot, the network continues to produce blocks
+but without including any transactions, and ensuring any fees paid are 0. This
+will run for a certain number of slots, after which the network will stop
+producing blocks. This will allow the network to stabilise and produce a new
+genesis ledger from the last ledger produced by the network.
+
+This feature enables part of this procedure, by adding the definition of the
+slots and the mechanisms to stop the node from processing transactions and to
+stop the networks after those slots.
+
+## Detailed design
+
+The procedure to stop processing transactions and producing/validating empty
+blocks after a certain slot will be as follows:
+
+* There will be a configuration parameter set at compile-time that will define
+  the slot at which the node will stop processing transactions.
+* The previous configuration cannot be overridable at runtime, by design, as
+  this compromises the safety of the daemon software.
+* The node (daemon) will stop accepting new transactions from clients after
+  the configured slot.
+* After the configured slot, the block producer will stop including transactions
+  in blocks, and all fees are set to 0.
```

Comment:
Can you be specific about what fees you are talking about here? Snark work fees? Block rewards?

Reply:
I'm pretty sure that this is referring to snark work fees, but the below section says that no snark work will be included. If the latter is true, can we update these sections talking about 0 fees to instead refer to no snark work? The logic about rejecting non-zero fees isn't accurate if the rule is instead that we reject any new completed works altogether.

Reply:
Agree. Clarified block production/validation sections to be specific on what blocks should/shouldn't include when they are produced/validated after the stop slot.